pepper-apis/src/main/java/org/broadinstitute/ddp/util/QuestionUtil.java

```diff
@@ -23,13 +24,20 @@ public static boolean isReadonly(Handle handle, QuestionDto dto, String activity
             return false;
         }
 
-        InstanceStatusType statusType = handle.attach(JdbiActivityInstance.class)
-                .getByActivityInstanceGuid(activityInstanceGuid)
-                .map(ActivityInstanceDto::getStatusType)
-                .orElseThrow(() -> new DaoException(String.format(
+        ActivityInstanceDto activityInstanceDto = handle.attach(JdbiActivityInstance.class)
+                .getByActivityInstanceGuid(activityInstanceGuid).orElseThrow(() -> new DaoException(String.format(
                         "Could not find activity instance with guid=%s while getting question with id=%d and stableId=%s",
                         activityInstanceGuid, dto.getId(), dto.getStableId())));
 
-        return InstanceStatusType.COMPLETE.equals(statusType);
+        InstanceStatusType statusType = activityInstanceDto.getStatusType();
+        if (InstanceStatusType.COMPLETE.equals(statusType)) {
+            return true;
+        }
+
+        // If it is not the first instance - block editing
+        var instanceDao = handle.attach(ActivityInstanceDao.class);
+        Long previousInstanceId = instanceDao.findMostRecentInstanceBeforeCurrent(activityInstanceDto.getId())
```

Comment:
If we have multiple questions in the same activity defined as write-once, then we're potentially doing this lookup multiple times even though it's the same instance id. I think it will be ideal if we do this lookup once and pass it in. However, looking through the code, I don't really see a good place for that. The refactor needed will be significant. So unless there's an easy way to minimize this lookup, I think the way you have it here is fine.